clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -1216,6 +1216,8 @@ static void handlePreferredName(Sema &S, Decl *D, const ParsedAttr &AL) {
 }
 
 bool Sema::isValidPointerAttrType(QualType T, bool RefOkay) {
+  if (T->isDependentType())
```

Comment:
Probably valuable to audit the other uses of this function well, and remove checks for dependence there as redundant.  And probably ensure the documentation of this function mentions that it skips on dependence.

Reply:
Done, please check the updated documentation of the function